### ISSUE TITLE
Add sherpa and gammapy among the dependencies (add them to install_requires in setup.py)

### ISSUE THREAD

**Issue by cosimoNigro** [cosimoNigro/agnpy#122]
So far we had only `numpy`, `scipy`, `astropy` and `matplotlib` among the dependencies.
These are specified in `setup.py` and automatically downloaded and installed when `agnpy` is installed via `pip` or `conda`.

After #119, I have introduced `gammapy` and `sherpa` wrappers for the `agnpy` radiative models.
This means that there are now classes in the new `agnpy.fit` submodule that are built on `sherpa` and `gammapy` objects.

Trying to simulate an installation from pip, I created a new conda environment.
```
mamba create -n agnpy_0.2.0_test python=3.9 
conda activate agnpy_0.2.0_test
```
I created a distribution (using the same command used by [the github action automatically publishing our releases on PyPI](https://github.com/cosimoNigro/agnpy/blob/master/.github/workflows/pip-upload.yml#L28)) and installed it
```
python setup.py sdist bdist_wheel
cd dist
pip install agnpy-0.2.0.tar.gz
```

When I try to import any object I get the following error:

```python
>>> from agnpy.emission_regions import Blob
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/cosimo/software/miniconda3/envs/agnpy_0.2.0_test/lib/python3.9/site-packages/agnpy/__init__.py", line 9, in <module>
    from .fit import *
  File "/Users/cosimo/software/miniconda3/envs/agnpy_0.2.0_test/lib/python3.9/site-packages/agnpy/fit/__init__.py", line 1, in <module>
    from .models import *
  File "/Users/cosimo/software/miniconda3/envs/agnpy_0.2.0_test/lib/python3.9/site-packages/agnpy/fit/models.py", line 1, in <module>
    from .gammapy_wrapper import (
  File "/Users/cosimo/software/miniconda3/envs/agnpy_0.2.0_test/lib/python3.9/site-packages/agnpy/fit/gammapy_wrapper.py", line 6, in <module>
    from gammapy.modeling import Parameter, Parameters
ModuleNotFoundError: No module named 'gammapy.modeling'
```

Curiously, just repeating the import a second time in the same python terminal does not raise any error
```python
>>> from agnpy.emission_regions import Blob
>>> blob = Blob()
>>> print(blob)
* spherical emission region
 - R_b (radius of the blob): 1.00e+16 cm
 - t_var (variability time scale): 4.13e-01 d
 - V_b (volume of the blob): 4.19e+48 cm3
 - z (source redshift): 0.07
 - d_L (source luminosity distance):9.92e+26 cm
 - delta_D (blob Doppler factor): 1.00e+01
 - Gamma (blob Lorentz factor): 1.00e+01
 - Beta (blob relativistic velocity): 9.95e-01
 - theta_s (jet viewing angle): 5.74e+00 deg
 - B (magnetic field tangled to the jet): 1.00e+00 G
 - xi (coefficient for 1st order Fermi acceleration) : 1.00e+00
* electron spectrum
 - power law
 - k_e: 9.27e+06 1 / cm3
 - p: 2.80
 - gamma_min: 1.00e+02
 - gamma_max: 1.00e+07
```

I think the only solution to avoid the error is to add `sherpa` and `gammapy` to the [`install_requires` in `setup.py`](https://github.com/cosimoNigro/agnpy/blob/master/setup.py#L23).
I do not think it is a big deal since these packages live basically all in the same environment.

On the other hand, if some users want to use `agnpy` for modelling, then they might not want to install `sherpa` and `gammapy` as well.

@jsitarek, any opinion?

@adonath, is there a way to skip the import of certain submodules (in this case of `agnpy.fit`)? If it is possible, would it be much more work than simply adding `sherpa` and `gammapy` to the dependencies?

**Comment by cosimoNigro** [comment]
CC @registerrier.

**Comment by jsitarek** [comment]
Hi @cosimoNigro 

if you want to avoid having those dependancies as required I guess you can put those gamma/sherpa imports in try/except block
```
try:
  import ...
except ImportError:
<here some error message that gammapy/sherpa is required and skip rest of the file >
```

nevertheless, sherpa and gammapy are already in the environment file provided with agnpy, so having them also included in the default setup should not be a big issue. I made a quick check of the size of environment without them (1.2 GB) and with them (some of the packages got downgraded slightly): 1.3GB

**Comment by cosimoNigro** [comment]
Thanks a lot for checking @jsitarek.
If the change in size of the environment is minimal, then I think I will just add `sherpa` and `Gammapy` to the dependencies and make the new release.
Will do the release tomorrow.

**Comment by cosimoNigro** [comment]
Solved by #125.
